R/say.r

```diff
@@ -101,12 +110,38 @@
 #' # Clippy
 #' say(fortune=59, by="clippy")
 
-say <- function(what="Hello world!", by="cat", type="message", length=18, 
-                fortune=NULL, ...) {
+say <- function(what="Hello world!", by="cat", 
+                type="message", 
+                what_color=NULL, by_color=NULL,  
+                length=18, fortune=NULL, ...) {
 
   if (length(what) > 1) {
     stop("what has to be of length 1", call. = FALSE)
   }
+  
+  if (!is.null(what_color) & !(inherits(what_color, c("crayon", "character")))) {
```

Comment:
singular `&` and `|`   are vectorized though so doesn't this result in two booleans? i assume you wanted one? (same with below)

Reply:
Ah guess I thought we'd need the single in order to be able to map over multiple arguments but I was wrong, this works fine. Not sure what's up with the extra NULLs at the end though. Doesn't happen with just a single animal

```
library(cowsay)

l <- 
  list(what = list("one", "two"), 
       by = list("cow", "chicken"),
       by_color = list(c("red", "blue"), c("green", "orange")), 
       what_color = "salmon")

purrr::pmap(l, say)
#>  ----- 
#> one 
#>  ------ 
#>     \   ^__^ 
#>      \  (oo)\ ________ 
#>         (__)\         )\ /\ 
#>              ||------w|
#>              ||      ||
#>  ----- 
#> two 
#>  ------ 
#>     \   
#>      \
#>          _
#>        _/ }
#>       `>' \
#>       `|   \
#>        |   /'-.     .-.
#>         \'     ';`--' .'
#>          \'.    `'-./
#>           '.`-..-;`
#>             `;-..'
#>             _| _|
#>             /` /` [nosig]
#> 
#> [[1]]
#> NULL
#> 
#> [[2]]
#> NULL
```